dream/utilities.py

```diff
@@ -78,28 +74,44 @@ def find_ndds_data_in_dir(
     assert os.path.exists(
         input_dir
     ), 'Expected path "{}" to exist, but it does not.'.format(input_dir)
+    dirlist = os.listdir(input_dir)
+
     assert isinstance(
         data_extension, str
     ), 'Expected "data_extension" to be a string, but it is "{}".'.format(
         type(data_extension)
     )
-    assert isinstance(
-        image_extension, str
-    ), 'Expected "image_extension" to be a string, but it is "{}".'.format(
-        type(image_extension)
-    )
+    data_full_ext = "." + data_extension
+
+    if image_extension is None:
+        # Auto detect based on list of image extensions to try
+        # In case there is a tie, prefer the first in this list
+        image_exts_to_try = ["png", "jpg"]
```

Comment:
what about jpeg, JPG. Or maybe hdr? I think in DOPE I have it as a parameter the user can specify.

Reply:
Good point. For cases where we convert a PNG-based NDDS dataset into JPEG, we can pick the extension, and thus 'jpg' would work (like we did with the datasets for the open-source release). If NDDS natively outputs something else besides 'jpg' when outputting JPEG images, that would be a reason to add it here... but I'm not sure what NDDS is set up to do in this case. I think I've always used it for generating PNG images.

I'm not sure about hdr images -- I hesitate adding the 'hdr' extension without testing the entire pipeline with hdr images.